### PR TITLE
Support for adding and removing rules, filters and actions through SubscriptionClient

### PR DESCRIPTION
**Pull request by nemakam**
## Description
Support for adding and removing rules, filters and actions through SubscriptionClient

Exposed the following methods in SubscriptionClient:
1. Task AddRuleAsync(string ruleName, Filter filter)
2. Task AddRuleAsync(RuleDescription description)
3. Task RemoveRuleAsync(string ruleName)

Currently supported filters: CorrelationFilter, SqlFilter, TrueFilter and FalseFilter
Currently supported action: SqlRuleAction

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs [hunk 1]
```diff
@@ -535,5 +565,52 @@ static Stream GetMessageBodyStream(AmqpMessage message)
...
+            }
+
+            correlationFilterMap[ManagementConstants.Properties.CorrelationFilterProperties] = propertiesMap;
+
+            return correlationFilterMap;
+        }
+
+        static AmqpMap GetRuleActionMap(SqlRuleAction sqlRuleAction)
+        {
+            AmqpMap ruleActionMap;
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs]: Could assign null to `ruleActionMap` and remove else statement?

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs [hunk 2]
```diff
@@ -46,5 +67,50 @@ protected override Task OnCloseAsync()
...
+            {
+                var amqpRequestMessage = AmqpRequestMessage.CreateRequest(
+                    ManagementConstants.Operations.AddRuleOperation,
+                    this.ServiceBusConnection.OperationTimeout,
+                    null);
+                amqpRequestMessage.Map[ManagementConstants.Properties.RuleName] = description.Name;
+                amqpRequestMessage.Map[ManagementConstants.Properties.RuleDescription] =
+                    AmqpMessageConverter.GetRuleDescriptionMap(description);
+
+                AmqpResponseMessage response =
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: `response` never used?

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs [hunk 3]
```diff
@@ -46,5 +67,50 @@ protected override Task OnCloseAsync()
...
+            try
+            {
+                var amqpRequestMessage =
+                    AmqpRequestMessage.CreateRequest(
+                        ManagementConstants.Operations.RemoveRuleOperation,
+                        this.ServiceBusConnection.OperationTimeout,
+                        null);
+                amqpRequestMessage.Map[ManagementConstants.Properties.RuleName] = ruleName;
+
+                AmqpResponseMessage response =
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: `response` not used?

File: src/Microsoft.Azure.ServiceBus/Filters/CorrelationFilter.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        {
+            if (value != null)
+            {
+                if (firstExpression)
+                {
+                    firstExpression = false;
+                }
+                else
+                {
+                    builder.Append(" And ");
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Filters/CorrelationFilter.cs]: Wonder if for consistency with [SQLFilter syntax](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-sql-filter), `AND` should be capitalized in the output.

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs (hunk 3 above)
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: Will use it for Exception handling in a separate PR

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs (hunk 2 above)
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: Will use it for Exception handling in a separate PR

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs (hunk 1 above)
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs]: (y)

File: src/Microsoft.Azure.ServiceBus/Filters/CorrelationFilter.cs (hunk 4 above)
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Filters/CorrelationFilter.cs]: (y)

**Pull request merged**